src/config.rs

```diff
@@ -744,45 +744,32 @@ impl<'a> Cfg<'a> {
         &self,
         verbose: bool,
     ) -> Result<(LocalToolchainName, ActiveReason)> {
-        match self.find_override_config()? {
-            Some((override_config, reason)) => match override_config {
-                OverrideCfg::PathBased(path_based_name) => {
-                    let toolchain = Toolchain::with_reason(self, path_based_name.into(), &reason)?;
-                    Ok((toolchain.name().clone(), reason))
-                }
-                OverrideCfg::Custom(custom_name) => {
-                    let toolchain = Toolchain::with_reason(self, custom_name.into(), &reason)?;
-                    Ok((toolchain.name().clone(), reason))
-                }
-                OverrideCfg::Official {
-                    toolchain,
-                    components,
-                    targets,
-                    profile,
-                } => {
-                    let toolchain = self
-                        .ensure_installed(&toolchain, components, targets, profile, verbose)
-                        .await?
-                        .1;
-                    Ok((toolchain.name().clone(), reason))
-                }
-            },
-            None => match self.get_default()? {
-                None => Err(no_toolchain_error(self.process)),
-                Some(ToolchainName::Custom(custom_name)) => {
-                    let reason = ActiveReason::Default;
-                    let toolchain = Toolchain::with_reason(self, custom_name.into(), &reason)?;
-                    Ok((toolchain.name().clone(), reason))
-                }
-                Some(ToolchainName::Official(toolchain_desc)) => {
-                    let reason = ActiveReason::Default;
-                    let toolchain = self
-                        .ensure_installed(&toolchain_desc, vec![], vec![], None, verbose)
-                        .await?
-                        .1;
-                    Ok((toolchain.name().clone(), reason))
-                }
-            },
+        if let Some((override_config, reason)) = self.find_override_config()? {
+            let toolchain = override_config.clone().into_local_toolchain_name();
```

Comment:
Since we're only getting LocalToolchainName now, calling `override_config.into_local_toolchain_name()` should be enough. There's no need for more complicated pattern matching other than checking whether the toolchain is official: if it is, we ensure its installation via `ensure_installed()`; if it isn't, we ensure that the override reason is valid via `Toolchain::with_reason()`.